java/arcs/core/storage/DirectStoreMuxer.kt

```diff
@@ -35,41 +33,80 @@ import kotlinx.coroutines.withContext
  */
 class DirectStoreMuxer<Data : CrdtData, Op : CrdtOperation, T>(
     val storageKey: StorageKey,
-    val backingType: Type,
-    val callbackFactory: (String) -> ProxyCallback<Data, Op, T>
+    val backingType: Type
 ) {
     private val storeMutex = Mutex()
     private val log = TaggedLog { "DirectStoreMuxer" }
+    private val proxyManager = RandomProxyCallbackManager<Data, Op, T>(
+        "backing",
+        Random
+    )
 
     // TODO(b/158262634): Make this CacheMap Weak.
     /* internal */ val stores = LruCacheMap<String, StoreRecord<Data, Op, T>>(
         50,
         livenessPredicate = { _, sr -> !sr.store.closed }
-    ) { _, sr -> closeStore(sr) }
+    ) { muxId, sr -> closeStore(muxId, sr) }
+
+    // Keeps track of all the stores a callbackId is registered to. Used when removing an
+    // observer (off method), as that requires unregistering to each store it is registered with.
+    private val callbackIdToMuxIdMap = mutableMapOf<Int, MutableSet<String>>()
+
+    fun on(callback: ProxyCallback<Data, Op, T>, callbackToken: Int? = null): Int {
```

Comment:
Since DirectStoreMuxer doesn't extend ActiveStore, do we even need the `callbackToken` param?

Reply:
The plan is to make DirectStoreMuxer extend ActiveStore in the near future (in fact, it may be my next change after I submit this PR).